src/hotspot/cpu/arm/assembler_arm.hpp

```diff
@@ -279,37 +279,31 @@ class VFP {
   class float_num : public fpnum {
    public:
     float_num(float v) {
-      _num.val = v;
+      _num_bits = PrimitiveConversions::cast<unsigned int>(v);
     }
 
-    virtual unsigned int f_hi4() const { return (_num.bits << 9) >> (19+9); }
-    virtual bool f_lo_is_null() const { return (_num.bits & ((1 << 19) - 1)) == 0; }
-    virtual int e() const { return ((_num.bits << 1) >> (23+1)) - 127; }
-    virtual unsigned int s() const { return _num.bits >> 31; }
+    virtual unsigned int f_hi4() const { return (_num_bits << 9) >> (19+9); }
+    virtual bool f_lo_is_null() const { return (_num_bits & ((1 << 19) - 1)) == 0; }
+    virtual int e() const { return ((_num_bits << 1) >> (23+1)) - 127; }
+    virtual unsigned int s() const { return _num_bits >> 31; }
 
    private:
-    union {
-      float val;
-      unsigned int bits;
-    } _num;
+    unsigned int _num_bits;
```

Comment:
The name _num_bits is confusing; I spent a few moments trying to figure out what it was the number
of bits of.  I think just _bits or _raw_bits or something like that would be better.  Similarly in double_num.
I also think the class names would be better with "num" => "bits", but that likely has fannout, so maybe
later (if at all).